go/pkg/symlinkopts/opts.go

```diff
@@ -0,0 +1,119 @@
+// Package symlinkopts provides an efficient interface to create unambiguous symlink options.
+package symlinkopts
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Options represents a set of options for handling symlinks.
+// To ensure a valid set of options, use one of the functions provided in this package.
+type Options uint32
+
+const (
+	Preserve Options = 1 << iota
+	NoDangling
+	IncludeTarget
+	Resolve
+	ResolveExternal
+)
+
+// String returns a string representation of the options.
+//
+// The string has at most five letters:
+//
+//	P for Preserve
+//	N for No Dangling
+//	I for Include Target
+//	R for Replace
+//	E for Replace External
+func (o Options) String() string {
+	var b strings.Builder
+	if o.Preserve() {
+		fmt.Fprint(&b, "P")
+	}
+	if o.NoDangling() {
+		fmt.Fprint(&b, "N")
+	}
+	if o.IncludeTarget() {
+		fmt.Fprint(&b, "I")
+	}
+	if o.Resolve() {
+		fmt.Fprint(&b, "R")
+	}
+	if o.ResolveExternal() {
+		fmt.Fprint(&b, "E")
+	}
+	return b.String()
+}
+
+// Preserve returns true if the options include the corresponding property.
+func (o Options) Preserve() bool {
+	return o&Preserve == Preserve
+}
+
+// NoDangling returns true if the options include the corresponding property.
+func (o Options) NoDangling() bool {
+	return o&NoDangling == NoDangling
+}
+
+// IncludeTarget returns true if the options include the corresponding property.
+func (o Options) IncludeTarget() bool {
+	return o&IncludeTarget == IncludeTarget
+}
+
+// Resolve returns true if the options include the corresponding property.
+func (o Options) Resolve() bool {
+	return o&Resolve == Resolve
+}
+
+// ResolveExternal returns true if the options include the corresponding property.
+func (o Options) ResolveExternal() bool {
+	return o&ResolveExternal == ResolveExternal
+}
+
+// ResolveAlways return the correct set of options to always resolve symlinks.
+//
+// This implies that symlinks are followed and no dangling symlinks are allowed.
+// Each target will have the path of the symlink.
```

Comment:
Not sure I understand what this means.

Reply:
Can you perhaps rephrase it so its more clear? Something like: Each symlink will be replaced by its target. For example, ....